src/utils/metrics/reporter.go

```diff
@@ -0,0 +1,120 @@
+package metrics
+
+import (
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"go.uber.org/zap"
+)
+
+// Reporter gathers metrics across jobs and reports them.
+// Concurrency-safe.
+type Reporter interface {
+	// WriteSummary dumps Reporter contents into the target.
+	WriteSummary()
+
+	SumBytesSent() uint64
+	ClearData()
+
+	// NewAccumulator returns a new metrics Accumulator for the Reporter.
+	NewAccumulator(jobID string) *Accumulator
+}
+
+// ZapReporter
+
+type ZapReporter struct {
+	data   ReporterData
+	logger *zap.Logger
+}
+
+// NewZapReporter creates a new Reporter using a zap logger.
+func NewZapReporter(logger *zap.Logger) Reporter {
+	return &ZapReporter{logger: logger}
+}
+
+func (r *ZapReporter) WriteSummary() {
+	stats, totals := r.data.SumAllStats()
+
+	r.logger.Info("stats", zap.Object("total", &totals), zap.Object("targets", stats))
+}
+
+func (r *ZapReporter) NewAccumulator(jobID string) *Accumulator {
+	if r == nil {
+		return nil
+	}
+
+	return newAccumulator(jobID, &r.data)
+}
+
+func (r *ZapReporter) ClearData() {
+	r.data.ClearData()
+}
+
+func (r *ZapReporter) SumBytesSent() uint64 {
+	return r.data.SumBytesSent()
+}
+
+// ConsoleReporter
+
+type ConsoleReporter struct {
+	data   ReporterData
+	target io.Writer
+}
+
+// NewConsoleReporter creates a new Reporter which outputs straight to the console
+func NewConsoleReporter(target io.Writer) Reporter {
+	return &ConsoleReporter{target: target}
+}
+
+func (r *ConsoleReporter) WriteSummary() {
+	w := tabwriter.NewWriter(r.target, 1, 1, 1, ' ', tabwriter.AlignRight)
```

Comment:
To be honest, I'm not exactly fond of this tab writer interface, especially since all the other output is done via single library. If the format doesn't suffer too much I'd rather have this reporter print stats via zap similar to how it was done before your previous PR that grouped the stats together

Reply:
Main motivation here is to simplify maintenance

Reply:
my motivation was - to make terminal output (if nobody actually logs) readable again - it's really hard with `zap` - they don't even support indentation for json. There are ways.. but it's like reimplementing a whole `Encoder`. Or wrapping. In short - I didn't find an easy way to pretty-print into something readable with `zap`. There are also external tools, but those are separate binaries and they won't be able to eat this simplified output anyway.

So the main problem for me personally is now - I can't tell anymore if load testing is going well or if all requests are blocked. That influenced my choice of VPNs, etc. That's why I was trying to resurrect the pretty table - it was soooo helpful with this. It would be just one method after all.

So, do you think I should go the hard way and attempt it with zap? It will probably be much more code.

Reply:
@Arriven 
I miss the table output as well. I run db1000n in docker containers, each one represents a separate VPN connection. It was relatively easy to print the logs and see the actual response rate for each instance. Having an option to print the table would be great (maybe introduce a new flag). Or maybe create a service, which can parse the output and display the stats (could be a docker container running a web server and a simple dashboard).

Reply:
yep, in my case no zap output with stats is ever readable. It was a bit more readable when every target was logged as a separate log entry, but it made more sense to combine the whole stats output into one entry for analysis, which I did.

"we output each target in a different log entry with each field being separate" - I think it was Before the change when I combined all stats into one entry, and you then brushed it up a bit and now it is something like this:
`logger.Info("stats", zap.Object("total", &totals), zap.Object("targets", stats))`

But even before combining all stats into one entry it was only somewhat readable for me - even with a separate log entry for each target. Nothing like the sweet table.

Reply:
I kept the table for now, after the separation the whole table reporter implementation is only ~+20 lines of code working with a tab writer, and it uses the separated metrics data storage for all data acquisition. I hope it's not such a big overhead for the feature many loved and used.. but if you insist - I'll remove it.